### PR TITLE
Fixed inventory handling.

### LINKED ISSUE #2035: Crafting issues.
I encountered various issues with crafting. 

I first tried drag-to spread and spread out my wood planks around the mini crafting slot in the player window. On the client side this appeared to work, but upon trying to get the crafting result it failed.

Then I tried manually right-clicking with the planks in all of the boxes. Upon clicking on the result, it produced 4 sticks instead of the expected crafting table, while still using the original amount of wood.

Finally, I tried the method in the paragraph above again, and it did produce a crafting table and I placed the table in my inventory. When I pressed escape to exit the window, another crafting table was dropped on the ground in front of me, but upon a relog the dupe was fixed.

### PR DESCRIPTION
**Pull request by madmaxoft**
Fixes #2035.

### CODE REVIEW

**Comment by worktycho** [comment]
Will this mean we need to reopen #1995?

**Comment by madmaxoft** [comment]
No, I fixed both at the same time. This is how it should have been fixed in the first place.

**Comment by worktycho** [comment]
Merge if CI succeeds then

File: src/Protocol/Protocol18x.cpp [hunk 1]
```diff
@@ -2265,7 +2265,7 @@ void cProtocol180::HandlePacketCreativeInventoryAction(cByteBuffer & a_ByteBuffe
 	{
 		return;
 	}
-	m_Client->HandleCreativeInventory(SlotNum, Item, (SlotNum == SLOT_NUM_OUTSIDE) ? caLeftClickOutside : caLeftClick);
+	m_Client->HandleCreativeInventory(SlotNum, Item, (SlotNum == -1) ? caLeftClickOutside : caLeftClick);
```
Comment (tigerw) [inline src/Protocol/Protocol18x.cpp]: But values are meant to be named?

**Pull request merged**